### PR TITLE
Use prot flags sent by client to map dma regions

### PR DESCRIPTION
**Pull request by swapnili**
fixes:#216
Signed-off-by: Swapnil Ingle <swapnil.ingle@nutanix.com>

### CODE REVIEW

File: lib/dma.h [hunk 1]
```diff
@@ -83,6 +83,9 @@ struct vfu_ctx;
 
 typedef struct {
     dma_addr_t dma_addr;        // DMA address of this region
+    uint32_t prot;              // memory protection of the mapping
```
Comment (tmakatos) [inline lib/dma.h:86]: It's `int` in `mmap(2)`.

File: lib/dma.c [hunk 2]
```diff
@@ -246,6 +246,12 @@ dma_controller_add_region(dma_controller_t *dma,
                        fd, offset, offset + size, region->fd);
                 goto err;
             }
+            if (region->prot != prot) {
+                vfu_log(dma->vfu_ctx, LOG_ERR, "bad prot=%#x "
+                        "for new DMA region %#lx-%#lx, existing prot=%#x\n",
+                        prot, offset, offset + size, region->prot);
```
Comment (tmakatos) [inline lib/dma.c:252]: Region should end at `offset + size - 1`.

**Review by tmakatos — APPROVED** [review]
Looks good, minor nits.

File: lib/dma.h (hunk 1 above)
Comment (swapnili) [reply lib/dma.h:86]: struct vfio_user_dma_region.prot is uint32_t

File: lib/dma.c (hunk 2 above)
Comment (swapnili) [reply lib/dma.c:252]: Ack

**Review by tmakatos — APPROVED** [review]

**Pull request merged**